src/ucp/rndv/rndv.c

```diff
@@ -421,11 +421,7 @@ ucs_status_t ucp_am_rndv_send_rts(ucp_request_t *sreq, uct_pack_callback_t pack_
 
     status = ucp_rndv_send_rts(sreq, pack_cb, rts_size);
     if (status == UCS_ERR_NO_RESOURCE) {
-        status = ucp_am_copy_header_to_internal_buf(sreq);
-        if (ucs_likely(status == UCS_OK)) {
-            return UCS_ERR_NO_RESOURCE;
-        }
-        return status;
+        return ucp_am_copy_header_to_internal_buf(sreq, status);
```

Comment:
seems like `status == UCS_ERR_NO_RESOURCE` is handled 2 times, see comment above